cw_monero/lib/monero_unspent.dart

```diff
@@ -7,27 +7,29 @@ class MoneroUnspent extends Unspent {
   MoneroUnspent(
       String address, String hash, String keyImage, int value, bool isFrozen, this.isUnlocked)
       : super(address, hash, value, 0, keyImage) {
+    getCoinByKeyImage(keyImage).then((coinId) {
+      if (coinId == null) throw Exception("Unable to find a coin for address $address");
+      getCoin(coinId).then((coin) {
+        _frozen = monero.CoinsInfo_frozen(coin);
+      });
+    });
   }
 
+  bool _frozen = false;
+
   @override
   set isFrozen(bool freeze) {
     printV("set isFrozen: $freeze ($keyImage): $freeze");
-    final coinId = getCoinByKeyImage(keyImage!);
-    if (coinId == null) throw Exception("Unable to find a coin for address $address");
-    if (freeze) {
-      freezeCoin(coinId);
-    } else {
-      thawCoin(coinId);
-    }
-  }
-
-  @override
-  bool get isFrozen {
-    printV("get isFrozen");
-    final coinId = getCoinByKeyImage(keyImage!);
-    if (coinId == null) throw Exception("Unable to find a coin for address $address");
-    final coin = getCoin(coinId);
-    return monero.CoinsInfo_frozen(coin);
```

Comment:
Shouldn't this still exist, but returning the new `_frozen` variable you made?

Reply:
correct, I've missed that